package/kedro_viz/server.py

```diff
@@ -317,9 +318,22 @@ def format_pipelines_data(pipelines: Dict[str, "Pipeline"]) -> Dict[str, list]:
     # keep track of node_id -> set(child_node_ids) for layers sorting
     node_dependencies = defaultdict(set)
     tags = set()
+    # keep track of modular pipelines
+    modular_pipelines = set()
 
     for pipeline_key, pipeline in pipelines.items():
         pipelines_list.append({"id": pipeline_key, "name": _pretty_name(pipeline_key)})
+        modular_pipelines.update(
+            set(
+                chain.from_iterable(
+                    [
+                        _expand_namespaces(x.namespace)
+                        for x in pipeline.nodes
+                        if x.namespace
+                    ]
+                )
```

Comment:
I think this can be made even more clear and concise:
```suggestion
modular_pipelines = list({y for y in _expand_namespaces(x.namespace)
                        for x in pipeline.nodes})
```